include/deal.II/base/array_view.h

```diff
@@ -157,6 +157,35 @@ class ArrayView
    */
   ArrayView (std::vector<typename std::remove_cv<value_type>::type> &vector);
 
+  /**
+   * Compare two ArrayView objects of the same type. Two objects are considered
+   * equal if they have the same size and the same starting pointer.
+   * This version always compares with the const value_type.
+   */
+  constexpr bool operator == (const ArrayView<const value_type> &other_view) const;
```

Comment:
Hm, that is interesting -- isn't it the case that you can only make this function `constexpr` if you only call `constexpr` functions inside? Which you don't -- you call `begin()` and `size()`.

Reply:
i also though so

Reply:
Yes, currently calling this in a constexpr call will fail but it works in a runtime call of course.

I want to make the whole class `constexpr` in a subsequent PR. Since we are using this object increasingly as temporary objects at interfaces this seems to be the right thing to do such that the compiler can eliminate it completely.

I am fine with changing the interface back temporarily such that the introduction is uniform.